subprojects/core/src/main/java/org/gradle/api/internal/project/taskfactory/AnnotationProcessingTaskFactory.java

```diff
@@ -47,6 +48,8 @@ public ITaskFactory createChild(ProjectInternal project, Instantiator instantiat
     private <S extends Task> S process(S task) {
         TaskClassInfo taskClassInfo = taskClassInfoStore.getTaskClassInfo(task.getClass());
 
+        ((TaskStateInternal) task.getState()).setIncremental(taskClassInfo.isIncremental());
```

Comment:
This name may be misleading for users. Just because the task takes IncrementalTaskInputs doesn't mean it's actually incremental on any given build. For instance, if there was not execution history or if the outputs have been changed, the task will have to rerun completely. What you are querying here is more like "is it incremental capable?"

The real incremental information is available in `IncrementalTaskInputs#isIncremental`.

Reply:
Point taken! I'll change it. 👍

Reply:
Done in https://github.com/gradle/gradle/pull/7849/commits/de208eacd27a0f551d5bff94da9ce6bd178066e5.